lib/heap_sort.rb

```diff
@@ -1,8 +1,24 @@
-
+require_relative "./min_heap"
 
 # This method uses a heap to sort an array.
-# Time Complexity:  ?
-# Space Complexity: ?
+# Time Complexity: On(log n)
+# Space Complexity: O(1)
 def heap_sort(list)
```

Comment:
You are building a heap here... so you DO have some space complexity even if that heap is empty by the end.